### PR TITLE
Api 31

### PR DESCRIPTION
**Pull request by hannesa2**
close https://github.com/owncloud/android/issues/3485

includes #3581

### CODE REVIEW

File: owncloudApp/build.gradle [hunk 1]
```diff
@@ -38,18 +38,17 @@ dependencies {
 
     implementation "androidx.lifecycle:lifecycle-extensions:2.2.0"
     implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$archLifecycleVersion"
-    implementation "androidx.lifecycle:lifecycle-livedata-ktx:2.3.1"
+    implementation "androidx.lifecycle:lifecycle-livedata-ktx:2.4.0"
     kapt "androidx.lifecycle:lifecycle-common-java8:$archLifecycleVersion"
 
-    kapt "org.xerial:sqlite-jdbc:3.34.0" // fix kapt for Apple Silicon https://stackoverflow.com/a/68285501/1079990
```
Comment (hannesa2) [inline owncloudApp/build.gradle:44]: There is no more need of it

**Comment by hannesa2** [comment]
It builds now properly, even local on Apple Silicon M1

**Review by abelgardep — APPROVED** [review]

**Comment by jesmrec** [comment]
Will check the impacting new features with API 31 and older to assure nothing is broken

||Android 12 | Android < 12|
-|-|-
|Custom Notifications | | Forcing an upload to fail, notification is correct. No way to collapse ¿? |
|Toast Redesign | | Toasts are being removed from app. Anyway, a check with the remaining ones was correct |
|Modern SameSite cookies in WebView | | 
|App hibernation|  No testable. System feature |  No testable. System feature |
|Safer component exporting | | | 
| Pending intents mutability | | | 
| Unsafe intent launches | | | 
| Foreground service launch restrictions | | | 
| Notification trampoline restrictions | | 
| Tests | | |

**Comment by jesmrec** [comment]
First appreciations:

[**Safer component exporting:**](https://developer.android.com/about/versions/12/behavior-changes-12#exported)

As the documentation states:

>If the app component includes the [LAUNCHER](https://developer.android.com/reference/android/content/Intent#CATEGORY_LAUNCHER) category, set android:exported to true. In most other cases, set android:exported to false.

every intent-filter `contains android:exported = true` even if it is not a LAUNCHER like `presentation.ui.sharing.ShareActivity` or `.authentication.AccountAuthenticatorService`. I guess that the true value is mandatory por LAUNCHER and for the other, it depends, right? 

[**Pending intents mutability:**](https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability)

our intents are always immutable, right? 

```
 val pendingIntent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_ONE_SHOT
        } else {
            PendingIntent.FLAG_ONE_SHOT
        }
```

could this affect the functionality somehow?

[**Unsafe intent launches**](https://developer.android.com/about/versions/12/behavior-changes-12#unsafe-intent-launches)

does the app do this? (as the docum): 

> Your app unparcels a nested intent from the extras of a delivered intent.<br> Your app immediately starts an [app component](https://developer.android.com/guide/components/fundamentals#Components) using that nested intent, such as passing the intent into [startActivity()](https://developer.android.com/reference/android/content/Context#startActivity(android.content.Intent)), [startService()](https://developer.android.com/reference/android/content/Context#startService(android.content.Intent)), or [bindService()](https://developer.android.com/reference/android/content/Context#bindService(android.content.Intent,%20android.content.ServiceConnection,%20int)).

[**Foreground service launch restrictions**](https://developer.android.com/about/versions/12/behavior-changes-12#foreground-service-launch-restrictions)

> Apps that target Android 12 or higher can't [start foreground services while running in the background](https://developer.android.com/guide/components/foreground-services#background-start-restrictions), except for [a few special cases](https://developer.android.com/guide/components/foreground-services#background-start-restriction-exemptions).

is any of the listed exceptions important to take in account?


**[non-SDK restrictions](https://developer.android.com/about/versions/12/behavior-changes-12#non-sdk-restrictions)**

anything that should be tested, regarding your comment -> https://github.com/owncloud/android/issues/3485#issuecomment-1070477790

**Comment by jesmrec** [comment]
QA Checks

- [X] Auto uploads in background (both photo and video)
    - [X] Regular uploads
    - [X] Only Wifi
    - [x] Only Charging 
- [X] Share with oC using different apps
- [X] Document provider (create folder, copy from, move to, delete, rename)
- [X] Upload/Download notifications 
- [X] E2E Automatic tests

**Comment by jesmrec** [comment]
I tried to put the focus in those features that use the system (checklist above), checking with different devices (Android 12 and older). Basic features work fine.

Since a new SDK could affect any part of the app, apart of the testing done in this PR, a [beta version ](https://github.com/owncloud/android/issues/3597) is recommended to expand the scope.

[Bumping the DAV and networking libraries](https://github.com/owncloud/android-library/issues/475) will help to fix https://github.com/owncloud/android/issues/3485#issuecomment-1070477790 

Extra testing will be done in regression stage.

This one is approved on my side. Thanks @hannesa2 and @fesave for the great work.

**Pull request merged**